packages/two_dimensional_scrollables/test/table_view/table_test.dart

```diff
@@ -1067,6 +1067,256 @@ void main() {
+        expect(verticalController.position.pixels, 0.0);
+        expect(horizontalController.position.pixels, 800.0);
+        // Max scroll extent was updated to reflect we no longer know where the
+        // end is, until the rowBuilder returns null again.
+        expect(verticalController.position.maxScrollExtent, 1400.0);
+        expect(horizontalController.position.maxScrollExtent, double.infinity);
+        // The layout should not have changed.
+        expect(find.text('R0:C5'), findsOneWidget);
+        expect(
+          tester.getRect(find.text('R0:C5')),
+          const Rect.fromLTRB(200.0, 0.0, 400.0, 200.0),
+        );
+        expect(find.text('R4:C7'), findsOneWidget);
+        expect(
+          tester.getRect(find.text('R4:C7')),
+          const Rect.fromLTRB(600.0, 800.0, 800.0, 1000.0),
+        );
+        // No columns laid out before column 3, but after column 7 we have added
+        // new columns.
+        expect(find.text('R0:C2'), findsNothing);
+        expect(find.text('R0:C8'), findsOneWidget);
+        // This exceeds the new bounds.
+        horizontalController.jumpTo(3200.0);
+        await tester.pumpAndSettle();
+        // Position was corrected.
+        expect(verticalController.position.pixels, 0.0);
+        expect(horizontalController.position.pixels, 2400.0);
+        // Max scroll extent was updated to reflect reaching the end of the
+        // columns after returning null again at the new index.
+        expect(verticalController.position.maxScrollExtent, 1400.0);
+        expect(horizontalController.position.maxScrollExtent, 2400.0);
+
+        // Decrease the number of columns
+        await tester.pumpWidget(MaterialApp(
+          home: getTableView(
+            rowCount: 10,
+            columnBuilder: (int index) {
+              // There will only be 5 columns.
+              if (index == 5) {
+                return null;
+              }
+              return largeSpan;
+            },
+          ),
+        ));
+        await tester.pumpAndSettle();
+
+        // The position should have changed.
+        expect(verticalController.position.pixels, 0.0);
+        expect(horizontalController.position.pixels, 200.0);
+        // Max scroll extent was updated to the new end we have corrected to.
+        expect(verticalController.position.maxScrollExtent, 1400.0);
+        expect(horizontalController.position.maxScrollExtent, 200.0);
+        // The layout updated.
+        expect(find.text('R0:C2'), findsOneWidget);
+        expect(
+          tester.getRect(find.text('R0:C2')),
+          const Rect.fromLTRB(200.0, 0.0, 400.0, 200.0),
+        );
+        expect(find.text('R4:C4'), findsOneWidget);
+        expect(
+          tester.getRect(find.text('R4:C4')),
+          const Rect.fromLTRB(600.0, 800.0, 800.0, 1000.0),
+        );
+        // No columns laid out after column 5.
+        expect(find.text('R0:C5'), findsNothing);
+      });
+
```

Comment:
Thanks for adding these additional test cases!

Reply:
It was a great catch! Thank *you*!